source/solvers/flow_control.cc

```diff
@@ -5,54 +5,54 @@
-                                 const double &                   dt,
-                                 const unsigned int &             step_number)
+FlowControl<dim>::calculate_beta(const double &      average_velocity_n,
+                                 const double &      dt,
+                                 const unsigned int &step_number)
 {
-  // Getting flow rate and area of the last time step.
-  flow_rate_n = flow_rate.first;
-  area        = flow_rate.second;
-
-  // (Only after step time 1)
-  // If flow is now reached with no force, the "no_force" variable is set to
-  // false. It may means that slowing down the flow with the pressure drop
-  // (no force applied) is ineffective (too big pressure drop) or the flow rate
-  // now reached the intended value without force.
-  // In case the flow rate is in the initial threshold but the pressure drop is
-  // too small to slow it down, the threshold factor decreases by 2.
-  // Otherwise, the flow rate may take a lot time to reached the flow rate value
-  // only with the pressure drop.
-  if (abs(beta_n - 0) < 1e-6 && step_number > 1)
+  double beta_n1;
+  // Check if disabling no_force and modify threshold factor (> step time 1):
+  // If average velocity target is reached without any beta force applied,
+  // the "no_force" is
+  // disabled. It means that the average velocity is now reached the
+  // target value without force OR that slowing down the flow with the only
+  // pressure drop (no force applied) is ineffective (too big pressure drop).
+  // In case the average velocity is in the initial threshold but the
+  // pressure drop is too small to slow down the flow, the threshold factor
+  // decreases by 2. Otherwise, the average velocity may take a lot time to
+  // reached the target value only with pressure drop.
+  if (abs(beta_n) < 1e-6 && step_number > 1)
     {
-      if (abs(flow_rate_n) < abs(flow_rate_0))
+      if (abs(average_velocity_n) < abs(average_velocity_0))
         no_force = false;
-      else if (abs(flow_rate_1n - flow_rate_n) <
-                 abs(flow_rate_n - flow_rate_0) &&
+      else if (abs(average_velocity_1n - average_velocity_n) <
+                 abs(average_velocity_n - average_velocity_0) &&
                threshold_factor > 1.00125)
-        threshold_factor = 1 + 0.5 * (threshold_factor - 1);
+        threshold_factor = 0.5 * (1 + threshold_factor);
     }
 
-  // (Only at step time 3)
-  // If flow rate is over the desired value at time step 1 and beta applied at
-  // time step 2 decreased it under the value, "no_force" is disable.
-  // As the previous condition, if flow rate is below the value after the small
-  // beta applied at time step 2, it means the pressure drop is too big. Then,
-  // slowing down the flow rate with the pressure drop is ineffective.
-  // This early disabling prevents to set a potential beta to 0 which could
+  // Check if disabling no_force is needed (step 3):
```

Comment:
I don't see the step 2, is it written explicitly?

Reply:
I removed this part, it wasn't really relevant anymore since we have more parameters to help flow control.